API_GUIDE.md

```diff
@@ -22,7 +22,67 @@ The [XLA readme](https://github.com/pytorch/xla/blob/master/README.md) describes
 
 ## Running a model
 
-To run a model, use the following API:
+There are different ways to run a model using the PyTorch/XLA framework.
+The simplest (but not good performing) one is to just run on one core and send the input tensors to the XLA devices manually:
+
+```
+import torch_xla_py.xla_model as xm
+
+device = xm.xla_device()
+model = MNIST()
+loss_fn = nn.NLLLoss()
+optimizer = optim.SGD(model.parameters(), lr=lr, momentum=momentum)
+
+model.train()
+for data, target in train_loader:
+  optimizer.zero_grad()
+  data = data.to(device)
+  target = target.to(device)
+  output = model(data)
+  loss = loss_fn(output, target)
+  loss.backward()
+  xm.optimizer_step(optimizer)
+```
+
+The above in only running on one TPU core though, and the the time spent to send data to device in serial/inline with the TPU computation.
+For simple experiments, or for inference tasks which do not latency sensitive it might be still OK, but the following methods allows for better scalability.
+
+Note the `xm.optimizer_step(optimizer)` line which replaces the usual `optimizer.step()`. This is required because of the way XLA tensors work: operations are not executed immediately, but rather added to a graph of pending operations which is only executed when its results are required. Using `xm.optimizer_step(optimizer)` acts as an execution barrier which forces the evaluation of the graph accumulated for a single step. Without this barrier, the graph would only be evaluated when evaluating the accuracy of the model, which is only done at the end of an epoch, for this example. Even for small models, the accumulated graph would be too big to evaluate at the end of an entire epoch.
+
+There are two ways to drive multiple TPU cores using PyTorch/XLA. One is using the `torch.multiprocessing` module (which internally spawns multiple processes), and the other is using Python threading.
```

Comment:
Could we structure `simplest` and two ways of using multi cores into subsections? 
Personally I feel it'll be more clear that way. ;) 
Btw if threading solution is less performant than the multiprocessing one, maybe we just want people to use that? 
I feel as a general guide it's more helpful to have:
- if you run your native Pytorch code, it works :D 
- if you want to squeeze out best performance running on TPU, please use multiprocessing.
So that users have a clear guideline on what to use. 
We can save the threading one somewhere else and not mention it here?

Reply:
Sure, I can make sub-sections.
WRT to the MT vs MP ... I feel like MP is too new to drop MT (which has been the one users started to use PyTorch/XLA with).